src/Mimi.jl

```diff
@@ -271,12 +278,25 @@ end
 function connectparameter(m::Model, target_component::Symbol, target_name::Symbol, source_component::Symbol, source_name::Symbol)
     c_target = m.components[target_component]
     c_source = m.components[source_component]
+
+    # Check the units, if provided
+    @assert unitcheck(getmetainfo(m, target_component).parameters[target_name].unit,
+                      getmetainfo(m, source_component).variables[source_name].unit) "Units of $source_component.$source_name do not match $target_component.$target_name."
+
     setfield!(c_target.Parameters, target_name, getfield(c_source.Variables, source_name))
     push!(m.parameters_that_are_set, string(target_component) * string(target_name))
     nothing
 end
 
 """
+Default string, string unit check function
+"""
+function unitcheck(one::Union{UTF8String, ASCIIString}, two::Union{UTF8String, ASCIIString})
+    # True if both "", either are "any", or they match
+    return one == "any" || two == "any" || one == two
```

Comment:
Could an empty string just mean "any"? I understand why we need something like "any", but I think I would prefer less of a special string...